### PR TITLE
Issue 10019: Fix embedding of media objects

### LINKED ISSUE #10019: Automatic media player embedding doesn't work for PeerTube, SoundCloud and other places
- [x] I have searched open and closed issues for duplicates

### Bug Description

If I share or embed a PeerTube video link, the resulting post only contains an empty frame in the middle along with the video link and description. The same happens for SoundCloud.

Embedding Vimeo or YouTube by URL doesn't work either, but they do work inside Vimeo and YouTube video tags **if given by ID**, respectively.

### Steps to Reproduce

1. Create a new post
2.  Click the "link or media" button at the bottom and paste the URL (or type in some [embedding BBCode](https://github.com/friendica/friendica/blob/46f9e9351e57a00ffa7f939704f37a61c425acab/doc/BBCode.md#embed) manually like `[video]` or `[audio]`)
3.  Press Share

Actual Result:

No video or audio player will appear, only an empty frame in the middle.

In case of YouTube or Invidious URL's, only a thumbnail will appear in place of a video player. While this is documented, it is unexpected behavior.

Expected Result:

I would expect an audio or video player to appear.

As a workaround, I could embed a deeplink to the mp4 file on the PeerTube instance and fill in the rich preview content by hand, but that's not acceptable for most users.

### Platform Info

Friendica Version: 2021.03-rc 

Friendica Source: https://nerdica.net/friendica 

PHP version:

SQL version:

### PR DESCRIPTION
**Pull request by annando**
Fixes #10019

Media objects are now embedded correctly.

### CODE REVIEW

**Comment by MrPetovan** [comment]
This PR reverts some of my work in #9929

The goal is for `/parse_url` to not allow arbitrary URL probing as much as possible. This is why the result is cached in `getInfoSiteCached`. `ParseUrl::getContentType` should not be used directly in a module we provide an arbitrary URL as a parameter.

File: src/Util/ParseUrl.php [hunk 1]
```diff
@@ -186,6 +210,12 @@ public static function getSiteinfo($url, $no_guessing = false, $do_oembed = true
 			return $siteinfo;
 		}
 
+		$type = self::getContentType($url);
```
Comment (MrPetovan) [inline src/Util/ParseUrl.php:209]: Ok, now we probably don't need lines 231-239 that were doing the same thing on the full request. This content type parsing was skipped for media files bigger than 1MB on line 227.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Util/ParseUrl.php (hunk 1 above)
Comment (annando) [reply src/Util/ParseUrl.php:209]: When testing I had significant speed impacts when fetching information for videos. For videos the system needed about 2-4 seconds to fetch information while `head` returns it immediately.

Comment (MrPetovan) [reply src/Util/ParseUrl.php:209]: I agree, but now lines 231-239 are superfluous and I can't remove them myself from the GitHub interface through suggestions.

**Review by MrPetovan — APPROVED** [review]

File: src/Util/ParseUrl.php (hunk 1 above)
Comment (annando) [reply src/Util/ParseUrl.php:209]: done - with some more changes.

**Pull request merged**